core/state/state_object.go

```diff
@@ -280,6 +300,16 @@ func (self *StateObject) Code(db trie.Database) []byte {
 }
 
 func (self *StateObject) SetCode(codeHash common.Hash, code []byte) {
+	prevcode := self.Code(self.db.db)
```

Comment:
It isn't possible to create an account, set code, delete it and set new code, therefor keeping a reference to the "old" code isn't required. `SetCode` can't be called twice for the same object.